client/verta/verta/endpoint/_endpoint.py

```diff
@@ -59,54 +68,99 @@ def __repr__(self):
-            "status: {}".format(status["status"]),
-            "date created: {}".format(data["date_created"]),
-            "date updated: {}".format(data["date_updated"]),
-            "stage's date created: {}".format(status["date_created"]),
-            "stage's date updated: {}".format(status["date_updated"]),
-            "components: {}".format(json.dumps(status["components"], indent=4)),
-        ))
+        return "\n".join(
+            (
+                "path: {}".format(data["creator_request"]["path"]),
+                "url: {}://{}/{}/endpoints/{}/summary".format(
+                    self._conn.scheme, self._conn.socket, self.workspace, self.id
+                ),
+                "id: {}".format(self.id),
+                "curl: {}".format(curl),
+                "status: {}".format(status["status"]),
+                "Kafka settings: {}".format(self.kafka_settings),
+                "date created: {}".format(data["date_created"]),
+                "date updated: {}".format(data["date_updated"]),
+                "stage's date created: {}".format(status["date_created"]),
+                "stage's date updated: {}".format(status["date_updated"]),
+                "components: {}".format(json.dumps(status["components"], indent=4)),
+            )
+        )
+
+    @property
+    def kafka_settings(self):
+        # NOTE: assumes endpoints only have one stage (true at time of writing)
+        return self._get_kafka_settings(self._get_or_create_stage())
 
     @property
     def path(self):
         return self._path(Endpoint._get_json_by_id(self._conn, self.workspace, self.id))
 
     def _path(self, data):
-        return data['creator_request']['path']
+        return data["creator_request"]["path"]
 
     def _date_updated(self, data):
-        return data['date_updated']
+        return data["date_updated"]
 
     @classmethod
-    def _create(cls, conn, conf, workspace, path, description=None, public_within_org=None, visibility=None):
-        endpoint_json = cls._create_json(conn, workspace, path, description, public_within_org, visibility)
+    def _create(
+        cls,
+        conn,
+        conf,
+        workspace,
+        path,
+        description=None,
+        public_within_org=None,
+        visibility=None,
+        kafka_settings=None,
+    ):
+        endpoint_json = cls._create_json(
+            conn, workspace, path, description, public_within_org, visibility
+        )
         if endpoint_json:
-            endpoint = cls(conn, conf, workspace, endpoint_json['id'])
-            endpoint._create_stage()  # an endpoint needs a stage to function, so might as well create it here
+            endpoint = cls(conn, conf, workspace, endpoint_json["id"])
+
+            # an endpoint needs a stage to function, so might as well create it here
+            stage_id = endpoint._create_stage()
+            if kafka_settings:
+                endpoint._set_kafka_settings(stage_id, kafka_settings)
```

Comment:
Here is where Kafka settings are set during `client.create_endpoint()`.

Notably not atomic, because our deployment API doesn't expose setting Kafka values during stage creation.